### PR TITLE
Update TransactionProcess according to API changes

### PR DESCRIPTION
**Pull request by robinsalehjan**
# Why?

- Add a new `style` property.

- Refactor from `secondaryButton` => `nativeButton` this button is used to replace steps where web uses `<a href>` in steps.

- `body` => `main` and `detail` remains. However these are now `object` types, hence why I introduced `TransactionStepContentViewModel `

# What?

- Introduce a `customStyle` property and enum for styling defined by the backend. This property is used to style the background of a step in case of warning/error.

- Both `main` and `detail` encapsulate a set of data represented by `TransactionStepContentViewModel`

Will follow up with another PR to have a more generic setup for `main` and `detail` content.

# Show me

| Warning | Error |
| --- | --- |
| Simulator Screen Shot - iPhone 11 Pro - 2020-03-27 at 15 45 01 | Simulator Screen Shot - iPhone 11 Pro - 2020-03-27 at 15 45 15 |

### CODE REVIEW

File: FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepViewModel.swift [hunk 1]
```diff
@@ -6,12 +6,20 @@ import Foundation
...
+    var main: TransactionStepContentViewModel? { get }
+    var detail: TransactionStepContentViewModel? { get }
+}
+
+public protocol TransactionStepContentViewModel {
+    var title: String? { get }
     var body: NSAttributedString? { get }
+    /*
+     For certain steps the attributed string assigned to body will contain a href element.
+     The host app will remove the <a href> element and the nativeButton will be rendered above the primaryButton.
```
Comment (osanoj) [inline FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepViewModel.swift]: Does this mean the app will need to parse the attributed string to remove any link elements? Or will the backend remove it?

Comment (robinsalehjan) [reply FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepViewModel.swift]: Yes, unfortunately the link element will be removed client side. I would like do this on the backend, but due to democracy my suggestion was not favoured 😞

Comment (robinsalehjan) [reply FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepViewModel.swift]: This is now redundant and will be fixed in the next PR.

File: FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+//
+//  Copyright © 2020 FINN AS. All rights reserved.
+//
+
+import Foundation
+
+public protocol TransactionStepContentViewModel {
+    var title: String? { get }
+    var body: NSAttributedString? { get }
+    /*
```
Comment (robinsalehjan) [inline FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentViewModel.swift]: Will change this

File: FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentView.swift [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    func transactionStepContentViewDidTapActionButton(
+        _ view: TransactionStepContentView,
+        inTransactionStep step: Int,
+        withAction action: TransactionStepContentView.ActionButton.Action,
+        withUrl urlString: String?,
+        withFallbackUrl fallbackUrlString: String?
+    )
+}
+
+public class TransactionStepContentView: UIView {
```
Comment (robinsalehjan) [inline FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentView.swift]: This is copy+paste from the `TransactionStepView`

File: FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentViewModel.swift (hunk 2 above)
Comment (robinsalehjan) [reply FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentViewModel.swift]: Introducing a new property `nativeBody` in addition to `body`. The difference is that `nativeBody` will only be provided by the backend if `body` contains the `<a href>` tag.

File: FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentView.swift (hunk 3 above)
Comment (robinsalehjan) [reply FinniversKit/Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepContentView/TransactionStepContentView.swift]: The `TransactionStepView` is a container and will render two instances of `TransactionStepContent` one for `main` and one for `detail`.

**Comment by robinsalehjan** [comment]
Opening a new PR since this became a little messy 😅

**Pull request closed without merging**